core/arch/arm/kernel/kern.ld.S

```diff
@@ -356,3 +378,8 @@ SECTIONS
 	}
 
 }
+
+#if defined(CFG_TEE_COHERENT_START)
+ASSERT((__coherent_start & (4096 - 1)) == 0, "Coherent RAM start is not 4Kbyte aligned")
+ASSERT((__coherent_end & (4096 - 1)) == 0, "Coherent RAM end is not 4Kbyte aligned")
```

Comment:
The assert that `. = ALIGN(4096);` works is just silly and and other assert could be done inside the `DECLARE_COHERENT_RAM_SECTION` macro instead.